### PR TITLE
Add update functions and docs

### PR DESCRIPTION
**Pull request by ehartley**
- I added a README with the instructions from https://github.com/mapping-commons/sssom/blob/master/src/docs/create-mapping-commons.md (with some updates) to this repo, so that it'll be easier to update them as I make changes to the repo. I also think it's a good idea to have the instructions here as well as in the SSSOM repo docs.

- In the `hooks/post_gen_project.py` file, I reformatted the commands into functions and added a `create_or_update_cruft_file()` function that helps with the process for updating the template variable values.  I also added the next step instructions (that should only be displayed upon project creation and not subsequent updates - though I'm not sure if my way of checking for this is the best method).

- In the `Makefile`, I added commands for `cruft check`, `cruft diff`, and `cruft update` so that these can be run in the ODK and I added a check to the `validate-mappings` target so that it doesn't fail when there are no mapping files.
 
- I made a few updates to the `{{cookiecutter.project_name}}/README.md` file and added the `{{cookiecutter.project_name}}/config/inverse_predicate_map.yml` file (to support mapping inversions).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+   git remote add origin https://gitlab.com/my-user-or-organization/my-commons-name.git
+   git add .
+   git commit -m 'Initial commit'
+   git push -set-upstream origin main
+   ```
+
+# What does this do?
+
+The following files and directories are autogenerated in the project:
+## TODO
```
Comment (matentzn) [inline README.md:76]:
```suggestion

## TODO

This section is under development. If you have questions about this in the meantime, reach out on https://github.com/mapping-commons/mapping-commons-cookiecutter/issues.
```

File: {{cookiecutter.project_name}}/.gitignore [hunk 2]
```diff
@@ -1,3 +1,2 @@
 tmp/
-*.pyc
-mappings/*.sssom.ttl
```
Comment (matentzn) [inline {{cookiecutter.project_name}}/.gitignore:3]: Why did you remove sssom.ttl from vc?  Maybe this is a good idea, but then we need to create the TTL files generated by the "validate-mappings" step in a different folder.

File: {{cookiecutter.project_name}}/config/inverse_predicate_map.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+inverse_predicate_map:
```
Comment (matentzn) [inline {{cookiecutter.project_name}}/config/inverse_predicate_map.yml:1]: Remind me about the use case of having this in the mapping commons as a separate file? Forgot it :D

**Review by matentzn — COMMENTED** [review]
Looks great! Thank you!

File: {{cookiecutter.project_name}}/config/inverse_predicate_map.yml (hunk 3 above)
Comment (ehartley) [reply {{cookiecutter.project_name}}/config/inverse_predicate_map.yml:1]: This file is one of the required inputs for the `sssom invert` function (https://mapping-commons.github.io/sssom-py/cli_usage.html#sssom-invert). Would it be better to add an "invert_mappings" option to the cookiecutter.json file and only include this file (and the yet-to-be-written corresponding invert commands in the Makefile) if the user selects "yes" for the option?

File: {{cookiecutter.project_name}}/.gitignore (hunk 2 above)
Comment (ehartley) [reply {{cookiecutter.project_name}}/.gitignore:3]: I've been going back and forth on this and welcome any suggestions. My first thought was that the sssom.ttl files were just auto-generated files created as one of the validation checks and so didn't need to be version controlled. Now for C-Path, we are ingesting the sssom.ttl files into the KG so we do need them to be stored in the online git repo. So I thought it might be better to have the default option be to include them in vc (hence the removal of the sssom.ttl files from `.gitignore`). One of my points of confusion regarding the sssom.ttl files is that they are created in the "validate-mappings" step.  Will the `sssom convert` command fail if there is an issue with the sssom.tsv file that doesn't get caught by the `sssom validate` command? Or is it that the sssom.ttl files should only be created after running `sssom validate`?

Comment (matentzn) [reply {{cookiecutter.project_name}}/.gitignore:3]: I think you are right to allow TTLs in the mapping folder. I would change the makefile that during mapping QC, the TTL files are generated in a temporary directory instead.

I included sssom convert in the validation pipeline because it catches some errors that other approaches cant catch, like illegal prefixing errors, or malformed entity identifiers..

File: {{cookiecutter.project_name}}/config/inverse_predicate_map.yml (hunk 3 above)
Comment (matentzn) [reply {{cookiecutter.project_name}}/config/inverse_predicate_map.yml:1]: I think maybe we should make this configurable as you say - I would like to keep the registry as lean as possible, and inversion maybe is only needed for a few select use cases.

File: {{cookiecutter.project_name}}/.gitignore (hunk 2 above)
Comment (ehartley) [reply {{cookiecutter.project_name}}/.gitignore:3]: I modified the `validate-%` command to output the .ttl files into the tmp directory.

File: {{cookiecutter.project_name}}/config/inverse_predicate_map.yml (hunk 3 above)
Comment (ehartley) [reply {{cookiecutter.project_name}}/config/inverse_predicate_map.yml:1]: I added the template variable `inverse_mappings` to cookiecutter.json with "No" as the default option. Now the inverse_predicate_map.yml file will only be added if `inverse_mappings = Yes`.

**Comment by ehartley** [comment]
@matentzn Can you review the updates I made?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+The following files and directories are autogenerated in the project:
+
+## TODO
+
+This section is under development. If you have questions about this in the meantime, reach out on https://github.com/mapping-commons/mapping-commons-cookiecutter/issues.
+
+
+# Future updates to the project's boilerplate code
+
+In order to be up-to-date with the template, first check if there is a mismatch between the project's boilerplate code and the template by running:
```
Comment (matentzn) [inline README.md]: Add a note that the python environment needs to be active to run this check?

**Review by matentzn — APPROVED** [review]
This looks great! I would vote for merging it and then iterating with the understanding that many things can still change.

File: README.md (hunk 4 above)
Comment (ehartley) [reply README.md]: Done. I updated this whole section with instructions on how to run the update commands in either the virtual environment or using the Makefile commands.
